Allow filtering by result on 'All tests' page

Martchus opened the pull request:
See https://progress.opensuse.org/issues/19388

I could not use the approach from the job results overview page. The
structure of the pages are just too different.

However, this implementation using the filtering of the data table
is quite simple and will not cause any performance regressions.

WIP because tests are missing. But maybe it is better to get some feedback before anyways.

![spectacle bw7714](https://user-images.githubusercontent.com/10248953/31185995-b07281be-a92d-11e7-90cd-1234d07726a6.png)

coolo commented:
I dislike another row just for a place holder

Martchus commented:
@coolo Yes, it is not so nice. At least when there is enough space, it should be in one row.

okurz commented:
How feasible would it be to also provide query parameters for fast access and shareable URLs?

Martchus commented:
@okurz It is possible to read query parameters also from JavaScript, so it could be done with this approach (which uses DataTable/JavaScript).

Martchus commented:
@okurz Actually it would be a one-liner (without parsing the query parameters), eg. `$('#finished-jobs-result-filter').val(["Softfailed", "Failed"]).trigger('chosen:updated').trigger('change')`.

@coolo No extra row anymore: ![spectacle z27565](https://user-images.githubusercontent.com/10248953/31228344-7b5b923a-a9dd-11e7-8147-b3a4303423c6.png)

I also added a test.

Martchus commented:
The CI fail is unrelated (only timeout). It passed earlier and all I amended was use of `$screen-sm` instead of hard-coded value.

okurz commented:
I recommend to merge the two commits. The CI jobs failing often recently is making me worry. I don't understand what changed to make the tests seemingly very unstable.

Martchus commented:
Ok, I'll squash it.

One of the CI failures is just a timeout:
```
Cannot read response header: timeout at /home/travis/perl_modules/lib/perl5/Devel/Cover/Report/Codecov.pm line 139.
```
This happened quite often lately. Maybe this could be prevented by splitting the tests further?

The fails due to "coverage change not visible in diff" are simply since we have the check for that.
